fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -383,17 +384,18 @@ def state_dict(self, *args, **kwargs):  # type: ignore
         .. warning:: This needs to be called on all ranks, since synchronization
             primitives will be used.
         """
-        torch.cuda.synchronize()
-        self._lazy_init()
```

Comment:
We still need these two lines to properly wait for the previous optimizer step to finish